pkg/terminal/command.go

```diff
@@ -97,11 +99,18 @@ See also: "help on", "help cond" and "help clear"`},
 		{aliases: []string{"trace", "t"}, cmdFn: tracepoint, helpMsg: `Set tracepoint.
 
 	trace [name] <linespec>
-	
+
 A tracepoint is a breakpoint that does not stop the execution of the program, instead when the tracepoint is hit a notification is displayed. See $GOPATH/src/github.com/derekparker/delve/Documentation/cli/locspec.md for the syntax of linespec.
 
 See also: "help on", "help cond" and "help clear"`},
-		{aliases: []string{"restart", "r"}, cmdFn: restart, helpMsg: "Restart process."},
+		{aliases: []string{"restart", "r"}, cmdFn: restart, helpMsg: `Restart process.
+
+  restart [-c checkpoint] [-args newargv...]
```

Comment:
Recordings can't get new arguments and live programs can't be restarted at a checkpoint, IMHO the -c/-args distinction is unnecessary and the arguments to `restart` can be interpreted contextually depending on what `client.Recorded()` returns.

Reply:
Ok, I changed restart to interpret the arg as the checkpoint when Recorded(). We still need to distinguish the case where (1) the user wants to restart with the current setting, and (2) the user wants to restart with an empty arg list. For (2) only, I retained the -args flag.